hoot-core/src/main/cpp/hoot/core/io/OsmGbdxJsonWriter.cpp

```diff
@@ -0,0 +1,391 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "OsmGbdxJsonWriter.h"
+
+// Boost
+using namespace boost;
+
+// Hoot
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/elements/ElementData.h>
+#include <hoot/core/elements/ElementType.h>
+#include <hoot/core/elements/Node.h>
+#include <hoot/core/elements/Relation.h>
+#include <hoot/core/elements/Tags.h>
+#include <hoot/core/elements/Way.h>
+#include <hoot/core/filters/NoInformationCriterion.h>
+#include <hoot/core/index/OsmMapIndex.h>
+#include <hoot/core/schema/OsmSchema.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/Exception.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/MetadataTags.h>
+#include <hoot/core/util/OsmUtils.h>
+#include <hoot/core/util/UuidHelper.h>
+#include <hoot/core/visitors/CalculateMapBoundsVisitor.h>
+
+// Qt
+#include <QBuffer>
+#include <QDateTime>
+#include <QFile>
+#include <QXmlStreamWriter>
+#include <QtCore/QStringBuilder>
+
+using namespace geos::geom;
+using namespace std;
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(OsmMapWriter, OsmGbdxJsonWriter)
+
+OsmGbdxJsonWriter::OsmGbdxJsonWriter(int precision)
+  : OsmJsonWriter(precision)
+{
+  _writeHootFormat = false;
+}
+
+void OsmGbdxJsonWriter::open(QString path)
+{
+  if (path.toLower().endsWith(".gbdx"))
+  {
+    path.remove(path.size() - 5, path.size());
+  }
+
+  _outputDir = QDir(path);
+  _outputDir.makeAbsolute();
+
+  if (_outputDir.exists() == false)
+  {
+    if (QDir().mkpath(_outputDir.path()) == false)
+    {
+      throw HootException("Error creating directory for writing.");
+    }
+  }
+}
+
+void OsmGbdxJsonWriter::_openFile()
+{
+  // Close the old file and open a new one
+  if (_fp.isOpen())
```

Comment:
I think I'd rather see an explicit close function used in this class, rather than have `open` closing previous files.

Reply:
I have kept the if open->close code and have renamed the function to _newOutputFile()